pennylane_lightning/src/bindings/Bindings.hpp

```diff
@@ -65,6 +65,33 @@ static auto create(pybind11::array_t<std::complex<PrecisionT>> &numpyArray)
         {data_ptr, static_cast<size_t>(numpyArrayInfo.shape[0])});
 }
 
+/**
+ * @brief Create a StateVectorManagedCPU object from a 1D numpy array
+ * by copying the internal data.
+ *
+ * @tparam PrecisionT Floating point precision type
+ * @param numpyArray Numpy array data-type
+ * @return StateVectorManagedCPU object.
+ */
+template <class PrecisionT = double>
+auto createManaged(
+    const pybind11::array_t<std::complex<PrecisionT>> &numpyArray)
+    -> StateVectorManaged<PrecisionT> {
+    pybind11::buffer_info numpyArrayInfo = numpyArray.request();
+
+    if (numpyArrayInfo.ndim != 1) {
+        throw std::invalid_argument(
+            "NumPy array must be a 1-dimensional array");
+    }
+    if (numpyArrayInfo.itemsize != sizeof(std::complex<PrecisionT>)) {
+        throw std::invalid_argument(
+            "NumPy array must be of type np.complex64 or np.complex128");
+    }
+    auto *data_ptr =
+        static_cast<std::complex<PrecisionT> *>(numpyArrayInfo.ptr);
+    return StateVectorManaged<PrecisionT>(
+        {data_ptr, static_cast<size_t>(numpyArrayInfo.shape[0])});
```

Comment:
We should be able to directly call the number of elements in the buffer with `size` as

```suggestion
        {data_ptr, static_cast<size_t>(numpyArrayInfo.size)});
```

Reply:
Updated!